lib/AST/ASTSearchTask.cpp

```diff
@@ -0,0 +1,58 @@
+//
+// Created by Stanislav Pankevich on 2019-10-19.
+//
+
+#include "mull/AST/ASTSearchTask.h"
+
+#include "mull/AST/ASTVisitor.h"
+#include "mull/Parallelization/Progress.h"
+
+using namespace mull;
+
+ASTSearchTask::ASTSearchTask(mull::ASTStorage &astStorage,
+                             mull::FilePathFilter &pathFilter,
+                             mull::TraverseMask traverseMask)
+    : astStorage(astStorage), pathFilter(pathFilter),
+      traverseMask(traverseMask) {}
+
+void ASTSearchTask::operator()(iterator begin, iterator end, Out &storage,
+                               mull::progress_counter &counter) {
+  mull::ASTMutations mutations;
+
+  for (auto it = begin; it != end; it++, counter.increment()) {
+    const std::unique_ptr<mull::Bitcode> &bitcode = *it;
+
+    const llvm::Module *const module = bitcode->getModule();
+
+    if (pathFilter.shouldSkip(module->getSourceFileName())) {
+      continue;
+    }
+
+    /// How can I check that -g flag (debug info enable) was set, from llvm pass
+    /// https://stackoverflow.com/a/21713717/598057
+    if (module->getNamedMetadata("llvm.dbg.cu") == nullptr) {
```

Comment:
I do believe that we need this kind of check, see #593. 

But maybe this could be moved even earlier, to be at the time when the bitcode has just finished loading.